release.py

```diff
@@ -515,9 +529,155 @@ def display_help_job_chain_for_source_calls(args):
+        _out = b""
+        _err = ""
+    else:
+        if _err:
+            print(f"An error happened running get_collections_from_package_and_version: {_err}")
+            sys.exit(1)
+
+    collection_list = _out.decode().strip().split(' ')
+    return collection_list
+
+
+def get_vendor_github_repo(package_name) -> str:
+    canonical_name = get_canonical_package_name(package_name)
+    return f"gazebo-release/{canonical_name.replace('-', '_')}_vendor"
+
+
+def get_vendor_repo_url(package_name) -> str:
+    return f"https://github.com/{get_vendor_github_repo(package_name)}"
+
+
+def prepare_vendor_pr_temp_workspace(package_name, ws_dir) -> Tuple[str, str, str]:
+    gz_vendor_tool = os.path.join(ws_dir, "gz_vendor")
+    # Create virtualenv for vendor dependencies
+    venv_dir = os.path.join(ws_dir, "venv")
+    venv.create(venv_dir, system_site_packages=True, with_pip=True)
+    subprocess.run([os.path.join(venv_dir, 'bin', 'pip3'), 'install', '-q',
+                    'jinja2==3.1.2',
+                    'catkin_pkg==1.0.0',
+                    'argparse'])
+    cmd = ['git', 'clone', '-q',
+           'https://github.com/gazebo-tooling/gz_vendor/',
+           gz_vendor_tool]
+    _, _err_tool = check_call(cmd, IGNORE_DRY_RUN)
+    gz_vendor_repo = os.path.join(ws_dir, 'gz_vendor_repo')
+    cmd = ['git', 'clone', '-q',
+           get_vendor_repo_url(package_name),
+           gz_vendor_repo]
+    _, _err_repo = check_call(cmd, IGNORE_DRY_RUN)
+    if _err_tool or _err_repo:
+        print("Problems with cloning vendor and tool repos:")
+        print(f"{_err_tool} {_err_repo}")
+        sys.exit(1)
+
+    return gz_vendor_tool, gz_vendor_repo, venv_dir
+
+
+def execute_update_vendor_package_tool(vendor_tool_path,
+                                       vendor_repo_path,
+                                       vendor_venv) -> None:
+    run_cmd = [os.path.join(vendor_venv, 'bin', 'python3'),
+               f"{vendor_tool_path}/create_gz_vendor_pkg/create_vendor_package.py",
+               'package.xml',
+               '--output_dir', vendor_repo_path]
+    _, _err_run = check_call(run_cmd)
+    if _err_run:
+        print("Problems running the create_vendor_package.py script:")
+        print(_err_run.decode())
+        sys.exit(1)
+
+
+def create_pr_for_vendor_package(args, repo_path, base_branch) -> str:
+    cmd_diff = ['git', "-C", repo_path, 'diff']
+    _out, _ = check_call(cmd_diff, IGNORE_DRY_RUN)
+    if not _out.decode():
+        return 'vendor tool did not produce any change, avoid the PR'
+
+    branch_name = f'releasepy/{args.version}'
+    vendor_repo = get_vendor_repo_url(args.package)
+    branch_cmd = ['git', "-C", repo_path,
+                  'checkout', '-b',  branch_name]
+    _, _ = check_call(branch_cmd, IGNORE_DRY_RUN)
+    commit_cmd = ['git', "-C", repo_path,
+                  'commit',
+                  '-m', f'Bump version to {args.version}',
+                  'CMakeLists.txt', 'package.xml']
```

Comment:
The tool might update other files as well in the future. I think we should commit all changes, not just to these two files.

Reply:
[c2da471](https://github.com/gazebo-tooling/release-tools/pull/1151/commits/c2da471d7fdb6a8c5c9770242ea806098099ffdb)